include/path_tracking_pid/controller.hpp

```diff
@@ -177,28 +179,28 @@ class Controller : private boost::noncopyable
   ControllerState getControllerState() const { return controller_state_; }
 
   // Set new vel_max_external value
-  void setVelMaxExternal(double value);
+  void setVelMaxExternal(units::Speed value);
 
   // Set new vel_max_obstacle value
-  void setVelMaxObstacle(double value);
+  void setVelMaxObstacle(units::Speed value);
 
   // Get vel_max_obstacle value
-  double getVelMaxObstacle() const;
+  units::Speed getVelMaxObstacle() const;
 
 private:
   void distToSegmentSquared(
     const tf2::Transform & pose_p, const tf2::Transform & pose_v, const tf2::Transform & pose_w,
-    tf2::Transform & pose_projection, double & distance_to_p, double & distance_to_w) const;
+    tf2::Transform & pose_projection, units::Area & distance_to_p,
+    units::Distance & distance_to_w) const;
 
   // Overloaded function for callers that don't need the additional results
-  double distToSegmentSquared(
-    const tf2::Transform & pose_p, const tf2::Transform & pose_v,
-    const tf2::Transform & pose_w) const
+  units::Area distToSegmentSquared(
+    const tf2::Transform & pose_p, const tf2::Transform & pose_v, const tf2::Transform & pose_w)
   {
     tf2::Transform dummy_tf;
-    double dummy_double = 0;
-    double result = 0;
-    distToSegmentSquared(pose_p, pose_v, pose_w, dummy_tf, result, dummy_double);
+    auto dummy_dist = units::Distance{0};
+    auto result = units::Area{0};
+    distToSegmentSquared(pose_p, pose_v, pose_w, dummy_tf, result, dummy_dist);
```

Comment:
In this case, the result is a bit odd. We're comparing squared distances only to avoid the `sqrt` on every one of them. Technically, that results in something with unit meter squared, but it does not represent any physical area.

Reply:
Yeah, I was in doubt as to what to call this. Since meters squared is the unit for area, I just went with that.

Can easily be renamed to `DistanceSquared` of course.

Reply:
Downside of that, of course, is that if you __do__ want to represent an actual area, this is still represented by a `DistanceSquared`. Otherwise calculations like `distance_1 * distance_2` become ambiguous.

Alternatively, you may want to introduce a type `Length`. Multiplying two `Length` type variables would become an `Area`, and multiplying two `Distance` type variables would become a `DistanceSquared`. Similar to the distinction between `Time` and `Duration`. However, in this case, it might become somewhat ambiguous to the user. Just brainstorming here...